test/tests.js

```diff
@@ -41,6 +41,12 @@ QUnit.test('percent character in cookie value mixed with encoded values', functi
 	assert.strictEqual(Cookies.get('bad'), 'foo%bar"baz%bax=', 'should read the percent character');
 });
 
+QUnit.test('percent character in cookie value mixed with encoded values lower case', function (assert) {
+	assert.expect(1);
+	document.cookie = 'mybad=foo%bar%22baz%bax%3d';
```

Comment:
No, I'm not aware of any browsers with differing behavior for encodeURIComponent.  However, it's not clear from your documentation that this library can only be used to read cookies that it creates.